Timezone/UserBasedTimezoneDetector.php

```diff
@@ -23,7 +23,7 @@
 class UserBasedTimezoneDetector implements TimezoneDetectorInterface
 {
     /**
-     * @var SecurityContextInterface
+     * @var SecurityContextInterface|TokenStorageInterface
      */
     protected $securityContext;
```

Comment:
This property should be deprecated too with getter/setter to be replaced by $tokenStorage.